store/tikv/coprocessor.go

```diff
@@ -656,15 +656,17 @@ func calculateRemain(ranges *copRanges, split *coprocessor.KeyRange, desc bool)
 }
 
 func (it *copIterator) Close() error {
-	close(it.finished)
+	if atomic.CompareAndSwapUint32(&it.closed, 0, 1) {
```

Comment:
Does the iterator will be closed multiple times?

Reply:
When `ctx.Done()` is called, `close(it.finished)` will be call here:
https://github.com/pingcap/tidb/pull/5701/files#diff-c27388ffb48c6f6eaeefe07dd3243530R443